pipelines/services/run.go

```diff
@@ -0,0 +1,187 @@
+package services
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+type RunService struct {
+	client *jfroghttpclient.JfrogHttpClient
+	auth.ServiceDetails
+}
+
+func NewRunService(client *jfroghttpclient.JfrogHttpClient) *RunService {
+	return &RunService{client: client}
+}
+
+const (
+	runStatus            = "api/v1/search/pipelines/"
+	triggerpipeline      = "api/v1/pipelines/trigger"
+	pipelineSyncStatus   = "api/v1/pipelineSyncStatuses"
+	pipelineResources    = "api/v1/pipelineSources"
+	syncPipelineResource = "api/v1/pipelineSources"
+	resourceVersions     = "api/v1/resourceVersions"
+)
+
+func (rs *RunService) GetRunStatus(branch, pipeName string, isMultiBranch bool) (*PipelineRunStatusResponse, error) {
+	httpDetails := rs.getHttpDetails()
+
+	// query params
+	queryParams := make(map[string]string, 0)
```

Comment:
I'm not sure about this one - 
If you know that the size is going to be 1, why initialize the capacity to 0?
I you don't know the capacity, you can remove the initial capacity initialization.
Same in the other places in the code (I counted 8)
```suggestion
	queryParams := make(map[string]string, 1)
```

Reply:
changed to declare the map to have all initial values instead of using length, capacity